Modules.md

```diff
@@ -0,0 +1,160 @@
+# Modules
+
+## Summary
+
+We propose adding a module system mechanism to the WGSL shading language as an extension.
+
+## Assumptions
+
+Assumes that [`load`](./Imports.md) has already been implemented.
+
+# Motivation
+
+Currently all symbols in wgsl share a single global namespace with a prohibition against symbols with the same name. Many existing implementations of imports (not unlike [our proposal](./Imports.md)) simply add to this global scope. For relatively self contained projects this is not a problem, however when considering a broader ecosystem of packages containing reusable shaders, collisions become much more likely. 
+
+Additionally WGSL provides very little way to encapsulate and organise code. This proposal would pave the way for both
```

Comment:
I think this is a keen observation. there's just no current way to describe a group of elements in wgsl right now!